src/adapter/src/config/frontend.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::time::Duration;
+
+use derivative::Derivative;
+use launchdarkly_server_sdk as ld;
+use tokio::time;
+
+use super::SynchronizedParameters;
+
+/// A frontend client for pulling [SynchronizedParameters].
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub struct SystemParameterFrontend {
+    #[derivative(Debug = "ignore")]
+    ld_client: ld::Client,
+    ld_user: ld::User,
+}
+
+impl SystemParameterFrontend {
+    /// Construct a new [SystemParameterFrontend] instance.
+    pub fn new(ld_sdk_key: &str, ld_user_key: &str) -> Result<Self, anyhow::Error> {
+        let ld_config = ld::ConfigBuilder::new(ld_sdk_key).build();
+        let ld_client = ld::Client::build(ld_config).map_err(|e| anyhow::format_err!(e))?;
```

Comment:
Confused about this `map_err`. Should be convertible to `anyhow::Error` without trouble?

```suggestion
        let ld_client = ld::Client::build(ld_config)?;
```